docs/src/developing/backwards-compatibility.md

```diff
@@ -155,3 +174,79 @@ circumvented in order to rapidly deploy a fix, depending on the severity of the
 CLI tooling json output (`output --json`) compatibility will be preserved; however, output directed
 for a human reader is subject to change. This includes output as well as potential help, warning, or
 error messages.
+
+### Why not just use SemVer
+
+The Solana Rust SDK crates use a model for introducing breaking changes that does
+*not* follow semantic versioning (SemVer).
+
+#### SemVer in Rust
+
+Under SemVer, breaking changes, such as removal of functions or types, only happens
+in a new major version.
+
+In many situations, this is useful -- it's very difficult to pick up a breaking
+change accidentally. The default dependency declaration in Cargo follows SemVer.
+It assumes that all minor versions are compatible, and will automatically update
+to the newest minor version available.
+
+For example, if you declare `solana-program = "1.10"`, Cargo can pull in version
+`1.16`, since it assumes that all `1.X.Y` releases are compatible.
+
+If, however, Cargo deems that two versions of a package are incompatible, it will
+treat them as two completely separate packages.
+
+Packages can be deemed incompatible for many reasons, but the most common situation
+under the default declaration format (SemVer) is due to different major versions.
+
+If a package ends up with two or more instances of `solana-program`,
+[bad things will happen](https://doc.rust-lang.org/cargo/reference/resolver.html#version-incompatibility-hazards),
+either at compile-time or runtime, if the conflicting types in the public APIs
+are ever used together.
+
+Because of these issues, developers should only want one instance of `solana-program`
+in their project.
+
+If developers all use the default versioning declaration with major version `1`,
+then we can ensure that only one version of `solana-program` exists in the resolution.
+
+The specifics of the Cargo resolver are beyond the scope of this document.
+You can find more information about the Cargo resolver in
+[The Cargo Book](https://doc.rust-lang.org/cargo/reference/resolver.html).
+
+#### So what's your solution?
+
+Rather than forcing developers to disentangle these kinds of messes on every
+major version release, our solution is to keep everything on the same major version.
+
+However, to avoid chaos and ensure that no downstream client ever breaks from updating
+from one `MINOR` version to the next, we must provide new functions or structs
+and deprecate the old ones.
+
+We have broken this commitment in the past, but through expanded downstream
+testing and conscientiously exposing APIs, we aim to never repeat that mistake.
+If it does happen accidentally, we will do whatever is necessary to restore
+the functionality, within reason.
```

Comment:
It is your call, you were working in this area for way longer than I did.
So, maybe it is a good thing to reference what happened in the past here.
But I personally would not include references to what happened in the past into the policy.

There is no magical way that always ensures 100% compatibility.
At least, I am not aware of one.
Except for not making any changes at all.

So the best we can do is the only thing we can do: we need to run tests to verify our promises.
And, I assume, everyone knows that test are not 100% reliable.
Test coverage matters and greatly affects the chances of a mistake, but it is, as everything else, is a subject of engineering time invested.
I think, we would be making an impossible commitment by saying "never".

I am not sure if policy needs to talk about the implementation details, but if we want to, maybe we could say something like:

```suggestion
We are in a process of setting up tests that verify compatibility guarantees to match
our policy.  But, as any test, it may have gaps andit is subject to engineering time
invested in the test coverage.  Tier 1 are major Solana ecosystem libraries, such as:

 * [Solana Program Library] (https://github.com/solana-labs/solana-program-library)
 * [Anchor](https://docs.rs/anchor-lang/latest/anchor_lang)
 * [Metaplex Token Library] (https://github.com/metaplex-foundation/metaplex-program-library)

Should we break our compatibility, we will do whatever is necessary to restore the
functionality, within reason.
```

I am actually not sure if the "Tier 1" list should be included here, nor if this is
the content for this list.  I just looked at the top dependencies of:
 * the `solana-program` crate: https://crates.io/crates/solana-program/reverse_dependencies
 * the `soalan-sdk` crate: https://crates.io/crates/solana-sdk/reverse_dependencies

Should we also include [Pyth](https://crates.io/crates/pyth-sdk-solana)?

Reply:
I would leave out the list and just explain that we're setting up tests. Much better, thanks!